File: 1813-maximum-erasure-value/maximum-erasure-value.cpp
class Solution {
public:
    int maximumUniqueSubarray(vector<int>& nums) {
        int start=0;
        int curr_sum=0,max_sum=0;
        unordered_map<int,int> mp;
        for(int end= 0;end<nums.size();end++){
            if(mp.find(nums[end])!=mp.end() && mp[nums[end]]>=start){
                while(start<=mp[nums[end]]){
                    curr_sum -= nums[start];
                    start++;
                }
            }
            mp[nums[end]] = end;
            curr_sum += nums[end];
            max_sum = max(curr_sum,max_sum);
        }
        return max_sum;
    }
};

File: 1813-maximum-erasure-value/maximum-erasure-value.py
class Solution:
    def maximumUniqueSubarray(self, nums: List[int]) -> int:
        d=dict()
        score = 0
        start = 0
        curr_sum=0
        for i,val in enumerate(nums):
            if val in d and d[val]>=start:
                while start<= d[val]:
                    curr_sum-=nums[start]
                    start+=1
            d[val] = i
            curr_sum += val
            score = max(score ,curr_sum)
        return score